### PR TITLE
feat(components): App guided tour

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

We need a high level component to handle guided tour & demo content import in apps

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { mount } from 'enzyme';
+import AppGuidedTour from './AppGuidedTour.component';
+import { LOADING_STEP_STATUSES } from '../Stepper';
+
+const DEFAULT_PROPS = {
+	appName: 'app name',
+	isOpen: true,
+	steps: [],
+	demoContentSteps: [],
+	onImportDemoContent: () => {},
```
Comment (frassinier) [inline packages/components/src/AppGuidedTour/AppGuidedTour.test.js:12]: Can you check callbacks too?

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+				onRequestClose();
+				setCurrentStep(0);
+			}}
+			steps={[
+				{
+					content: {
+						header: appName,
+						body: () => (
+							<div>
+								{t('GUIDE_TOUR_WELCOME_STEP_BODY', {
```
Comment (frassinier) [inline packages/components/src/AppGuidedTour/AppGuidedTour.component.js:62]:
```suggestion
								{t('GUIDED_TOUR_WELCOME_STEP_BODY', {
```

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+									defaultValue:
+										"If you're new, you may want to take a quick tour of the tool now. " +
+										'If not, you can replay the tour from the user menu.',
+								})}
+								<form>
+									<Toggle
+										label={t('GUIDE_TOUR_IMPORT_DEMO_CONTENT', {
```
Comment (frassinier) [inline packages/components/src/AppGuidedTour/AppGuidedTour.component.js:69]:
```suggestion
										label={t('GUIDED_TOUR_IMPORT_DEMO_CONTENT', {
```

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+										checked={importDemoContent}
+									/>
+								</form>
+							</div>
+						),
+					},
+				},
+				importDemoContent && {
+					content: {
+						header: t('GUIDE_TOUR_WELCOME_STEP_HEADER', {
```
Comment (frassinier) [inline packages/components/src/AppGuidedTour/AppGuidedTour.component.js:84]:
```suggestion
						header: t('GUIDED_TOUR_WELCOME_STEP_HEADER', {
```

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	);
+}
+
+AppGuidedTour.propTypes = {
+	isOpen: PropTypes.bool.isRequired,
+	appName: PropTypes.string.isRequired,
+	steps: GuidedTour.propTypes.steps.isRequired,
+	demoContentSteps: Stepper.propTypes.steps.isRequired,
+	onImportDemoContent: PropTypes.func.isRequired,
+	onRequestClose: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (gjou-tlnd) [reply packages/components/src/AppGuidedTour/AppGuidedTour.component.js:103]: yes, all of them

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js (hunk 1 above)
Comment (gjou-tlnd) [reply packages/components/src/AppGuidedTour/AppGuidedTour.test.js:12]: what do you mean?

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (gjou-tlnd) [reply packages/components/src/AppGuidedTour/AppGuidedTour.component.js:103]: I guess i can deal with null demoContentSteps

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js (hunk 1 above)
Comment (frassinier) [reply packages/components/src/AppGuidedTour/AppGuidedTour.test.js:12]: using jest.fn and be sure that there are called 😅

File: packages/components/src/AppGuidedTour/AppGuidedTour.component.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+				setCurrentStep(0);
+			}}
+			steps={[
+				{
+					content: {
+						header: appName,
+						body: () => (
+							<div>
+								{t('GUIDE_TOUR_WELCOME_STEP_BODY', {
+									defaultValue:
```
[the remark this answers is not in the document]
Comment (romainseb) [reply packages/components/src/AppGuidedTour/AppGuidedTour.component.js:63]: The contatenation with + is not working, we had issues on TDC side with that

File: packages/components/src/AppGuidedTour/AppGuidedTour.test.js (hunk 1 above)
Comment (gjou-tlnd) [reply packages/components/src/AppGuidedTour/AppGuidedTour.test.js:12]: I do ;)

Comment (gjou-tlnd) [reply packages/components/src/AppGuidedTour/AppGuidedTour.test.js:12]: I guess you meant it for onRequestClose, I'll add a test

**Pull request merged**